app-frontend/src/app/components/projects/projectSelectModal/projectSelectModal.module.js

```diff
@@ -26,38 +26,82 @@ class ProjectSelectModalController {
     }
 
     fetchPage(page = 1, search = this.search) {
+        if (this.resolve.isLayerSelect) {
```

Comment:
This looks like another opportunity to create a simpler generic component contained inside more targeted components with stuff like the fetch functions etc delegated to the containers.